### PR TITLE
mdns: Add APIs for looking up self hosted services and getting the self hostname (IDFGH-10130)

### PR DESCRIPTION
**Pull request by wqx6**
We need the APIs to lookup self hosted services and get the self hostname.

### CODE REVIEW

File: components/mdns/include/mdns.h [hunk 1]
```diff
@@ -128,6 +128,18 @@ void mdns_free(void);
  */
 esp_err_t mdns_hostname_set(const char *hostname);
 
+/**
+ * @brief Get the hostname for mDNS server
+ *
+ * @param hostname      pointer to the hostname
```
Comment (david-cermak) [inline components/mdns/include/mdns.h:134]: Could probably say that this `pointer to the hostname` needs to be allocated and hold at least `MDNS_NAME_BUF_LEN` chars.

**Review by david-cermak — APPROVED** [review]
LGTM

**Comment by david-cermak** [comment]
@wqx6 Please install the pre-commit hook and fix the commit message.

File: components/mdns/include/mdns.h (hunk 1 above)
Comment (wqx6) [reply components/mdns/include/mdns.h]: Added.

**Pull request merged**